doc/specs/#120,121,119-Update-Uninstall-List.md

```diff
@@ -0,0 +1,230 @@
+
+
+### Installed with Windows Package Manager
+If the app is installed with the Windows Package Manager, we have the ID to map the installed version to the repository version.  If the version installed on the PC is lower than the the version in the repository, the list command will show the available version in the "available update."
+
+### Installed out of band from the Windows Package Manaager but in repo
+If the app is installed out of band from the Package Manager, the app will be discovered if it registers correctly with the **Apps and Features**.  The Windows Package Manager will map the installed app back to the repo.  If the version installed is less than the version available in the repo, the **list** command will show the available version in the "available update."
+
+If the installed version is greater than or equal to the version of the app available in the repo, then the "available update" will be left blank.
+
+### Installed out of band from the Windows Package Manaager but not in repo
+If the app is installed out of band from the Package Manager, and is not available in the repo, then the Windows Package Manager will treat it as if there are no updates available.
+
+
+
+
+## Mapping between Windows Package Manager repo and Apps and Features
+*StoreApps*  
+Store apps will be mapped via the The Package Family Name (which is composed by Publisher and Name).   The Package Family Name will be consistent between Apps and Features and the Windows Package Manager repo
+
+*MSIs and EXEs*  
+Will be mapped to the Windows Package Manager repo via the SystemAppId.  The SystemAppId is a required field.  The value in the field differs depending on the InstallerType.
+
+For MSI it is the product code.  Typically a GUID that is typically found in the uninstall registry location and includes the brackets.
+For example: ```{5740BD44-B58D-321A-AFC0-6D3D4556DD6C}```
+
+```HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{3740BD44-B58D-321A-AFC0-6D3D4556DD6C}]```
+    
+```HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\{3740BD44-B58D-321A-AFC0-6D3D4556DD6C}]```
+
+For inno, wix, nullsoft, and exe, the SystemAppId should be a string that is located in either of the Uninstall keys above. 
+
+### Upgrade
+
+The **upgrade** command is designed to update one or more applications.  The **upgrade** command when executed with no parameters or with --all will update all packages that have updates available.
+
+
+## Usage
+
+`winget upgrade [[-q] \<query>] [\<options>]`
+
+![show command]()
+
+**Upgrade** when executed by itself will upgrade all apps ready for updates by the Windows Package Manager.
+
+
+| Argument  | Description |
+|--------------|-------------|
+| **-q,--query** |  The query used to search for an application. |
+| **-?, --help** |  Gets additional help on this command. |
+
+## Options
+
+The following options are available.
+
+| Option  | Description |
```

Comment:
What about specifying a manifest file that is located on the user's machine (or another repository)? Would this be part of another spec?